tracer/src/Datadog.Trace/Util/EnvironmentHelpers.cs

```diff
@@ -91,5 +91,22 @@ public static IDictionary GetEnvironmentVariables()
 
             return new Dictionary<object, object>();
         }
+
+        /// <summary>
+        /// Safe wrapper around Environment.GetEnvironmentVariable
+        /// </summary>
+        /// <param name="key">Name of the environment variable to fetch</param>
+        /// <param name="defaultValue">Value to return in case of error</param>
+        /// <returns>The value of the environment variable, or the default value if an error occured</returns>
+        public static bool GetBooleanEnvironmentVariable(string key, bool defaultValue)
+        {
+            var value = GetEnvironmentVariable(key);
+            return value switch
+            {
+                null => defaultValue,
+                "1" or "true" or "True" or "TRUE" => true,
+                _ => false
+            };
+        }
```

Comment:
We already have a (ridiculously optimized) `ToBoolean()` extension method for strings. We try to use that one everywhere for consistency (it's case-insensitive and also accepts `yes`, `no`, `y`, `n`, `T`, and `F`, which this ones doesn't). Also this version returns `false` for any unknown string.

See https://github.com/DataDog/dd-trace-dotnet/blob/0d5e185c59a46bd4702b8086eb08ddf79ef34605/tracer/src/Datadog.Trace/ExtensionMethods/StringExtensions.cs#L38

We could replace this implementation or remove it altogether:
```suggestion
        public static bool GetBooleanEnvironmentVariable(string key, bool defaultValue)
        {
            return GetEnvironmentVariable(key)?.ToBoolean() ?? defaultValue
        }
```

Reply:
Fixed in 18c25b94dbc2afc43c86450a638b85de37b6d58e